### DISCUSSION TITLE
Can m2e configure the warnings of the Eclipse compiler?

### DISCUSSION THREAD

**Discussion by trandersen-ufst** [eclipse-m2e/m2e-core#742] [Q&A]
I have a situation with a pure Maven multi-module project, where two of the modules generate Java code from an openapi generator.  This works well.  (Pure means we do not have any IDE specific files checked into git).

Unfortunately the code generated works, but has a lot of warnings like unused imports, and deprecated methods, which pollutes the Problem panel in Eclipse.  

I've managed to make the module compile from the command line with the Eclipse compiler and disabled the warnings with this pom.xml-snippet:

```xml
            <plugin>
                <!-- https://stackoverflow.com/questions/33164976/using-eclipse-java-compiler-ecj-in-maven-builds -->
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.10.1</version>
                <configuration>
                    <compilerId>eclipse</compilerId>
                    <showWarnings>true</showWarnings>
                    <compilerArguments>
                        <!-- https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_batch_compiler.htm&cp%3D1_3_8_0 -->
                        <!-- hack: use -showversion as it does not take an argument so "-warn..." is parsed as an option instead -->
                        <showversion>-warn:-serial,unusedImport,deprecation,unused</showversion>
                    </compilerArguments>

                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.codehaus.plexus</groupId>
                        <artifactId>plexus-compiler-eclipse</artifactId>
                        <version>2.12.0</version>
                    </dependency>

                    <dependency>
                        <groupId>org.eclipse.jdt</groupId>
                        <artifactId>ecj</artifactId>
                        <version>3.29.0</version>
                    </dependency>
                </dependencies>
            </plugin>
```

As m2e is capable of setting the compiler level, I was wondering if it also was capable of setting the project specific compiler options from data in the pom.xml.  I have not found documentation indicating it can, so this might be a nice enhangement if not.

If so, how should I do it?

**Comment by guillaumecle** [comment]
I would be interested in this feature as well.
In our pom we configure some eclipse compiler options. It would be nice if the IDE would automatically import that configuration.

https://github.com/hotpads/datarouter/blob/master/datarouter-parent/pom.xml
```xml
<compilerArgs>
<!-- https://git.eclipse.org/c/jdt/eclipse.jdt.core.git/tree/org.eclipse.jdt.core/batch/org/eclipse/jdt/internal/compiler/batch/Main.java#n4116 -->
	<arg>-err:+enumSwitch</arg>
	<arg>-err:+deadCode</arg>
	<arg>-err:+noEffectAssign</arg>
	<arg>-err:+serial</arg>
	<arg>-err:+staticReceiver</arg>
	<arg>-err:+suppress</arg>
	<arg>-err:+unlikelyCollectionMethodArgumentType</arg>
	<arg>-err:+unlikelyEqualsArgumentType</arg>
	<arg>-err:+unusedTypeArgs</arg>
</compilerArgs>
```

**Reply by mickaelistria** [reply]
That's a good proposal. Can you please create an enhancement request for this? And maybe consider submitting some code that fixes it, in org.eclipse.m2e.jdt. I don't think it would be too hard to implement that as it's mostly about setting the right preferences; but the difficult part is more the efficiently derive the preference to set from the argument as I'm not aware whether JDT does provide a good API/mapping for that.

**Reply by trandersen-ufst** [reply]
@mickaelistria I could make an enhancement request, but I do not have enough JDT experience to easily provide an initial implementation.   It is most likely necessary to provide custom code for the mapping.  Personally I would not mind having a very simple mapping like directly to the internal naming if that allowed to be used regardless of the compiler backend in use.     Is there perhaps a more experienced JDT programmer that could provide a proof-of-concept implementation to finish?

**Reply by trandersen-ufst** [reply]
@mickaelistria I have opened https://github.com/eclipse-m2e/m2e-core/issues/755

**Comment by mickaelistria** [comment]
Note that as per https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#compilerArgs is a List, not a name, so the name of the tag doesn't matter. So you called it `showversion` but `arg` or `doesNotMatter` would result in the same behavior, it does not matter in the end and only the value of the text element is passed to compiler, not the tag name.
